File: reverse/reverse.js
function reverse_number(n){
  const a = n.toString().split('').reverse().join('');
  return parseInt(a) * Math.sign(n);
}

// STEPS EXPLAINED
// Make the number a string
// Convert the String to an Array and Reverse the Array
// Convert the Reversed Array to a String
// Make the String a number
// Apply the Sign using Math.sign for +ve or -ve integers

File: reverse/reverse.dart
int reverse(int n)
{
 int rev=0;
  while(n!=0)
  {
    int p=n%10;
    int q=n-p;
    rev=rev*10+p;
    n= (q/10).floor();
  }
  return rev;
  
}
void main()
{
  int r=reverse(87852);
  print('reverse ${r}');
}

